src/repl/scala/tools/nsc/interpreter/InteractiveReader.scala

```diff
@@ -46,6 +46,13 @@ object InteractiveReader {
     }
 
   def apply(): InteractiveReader = SimpleReader()
+
+  // a non-interactive SimpleReader that returns the given text
+  def apply(text: String): InteractiveReader = SimpleReader(
```

Comment:
The owing of beers is all mine! I'm probably overthinking it. By the way, I've just started work on reorganizing the repl codebase to simplify maintenance and stealing of features from ammonite. My goal for the month is some more separation of concerns. Will open a ticket to discuss once I get a sense of the concrete direction I'm headed.